include/seqan3/search/algorithm/search.hpp

```diff
@@ -46,12 +46,27 @@
 namespace seqan3
 {
 
-//!\brief \todo Document!
+/*!\brief Search a query or a range of queries in an index.
+ * \param[in] index String index to be searched.
+ * \param[in] queries A single query or a (forward) range of queries. A query must model
+                      std::ranges::RandomAccessRange.
+ * \param[in] cfg A configuration object specifying the search parameters (e.g. number of errors, error types,
+ *                output format, etc.).
+ *
+ * ### Complexity
+ *
+ * Exponential in the numbers of errors.
+ *
+ * ### Exceptions
+ *
+ * No-throw guarantee.
```

Comment:
see above comments regarding complexity and exceptions. Also make statements about thread-safety

Reply:
I added a basic exception guarantee. Since queries might not be const (if it might be a range with a state), we cannot provide a strong exception guarantee.

Reply:
You can write exactly this, e.g. "Strong exception guarantee if iterating the query does not change its state; basic exception guarantee otherwise."